server/lib/wsapi/chat.ts

```diff
@@ -43,6 +28,31 @@ const sendThrottle = createThrottle('chatsend', {
   window: 60000,
 })
 
+function isChatServiceError(error: Error): error is ChatServiceError {
+  return error.hasOwnProperty('code')
+}
+
+function convertChatServiceError(err: Error) {
+  if (!isChatServiceError(err)) {
+    throw err
+  }
+
+  switch (err.code) {
+    case ChatServiceErrorCode.UserOffline:
+      throw new errors.NotFound(err.message)
+    case ChatServiceErrorCode.InvalidJoinAction:
+    case ChatServiceErrorCode.InvalidLeaveAction:
+    case ChatServiceErrorCode.InvalidSendAction:
+    case ChatServiceErrorCode.InvalidGetHistoryAction:
+    case ChatServiceErrorCode.InvalidGetUsersAction:
+      throw new errors.BadRequest(err.message)
```

Comment:
we *really* need to do the error body thing sooner rather than later :) (Although I guess for bad requests it's not as bad)